add flux kontext

tianyuan211 opened the pull request:
add flux kontext related files

tianyuan211 committed:
add flux kontext related files

tianyuan211 committed:
Update pipeline_flux_kontext.py

huijuanzh commented on optimum/habana/diffusers/pipelines/flux/pipeline_flux_kontext.py:
```diff
@@ -0,0 +1,712 @@
+        )
+        self._num_timesteps = len(timesteps)
+
+        # handle guidance
```
https://github.com/huggingface/diffusers/blob/v0.35.2/src/diffusers/pipelines/flux/pipeline_flux_kontext.py#L1013   why this pipeline do not have negative_ip_adapter_image related code. please check the pipeline code with original pipeline_flux_kontext.

huijuanzh commented on optimum/habana/diffusers/pipelines/flux/pipeline_flux_kontext.py:
```diff
@@ -0,0 +1,712 @@
+            import copy
+
+            transformer_bf16 = copy.deepcopy(self.transformer).to(self._execution_device)
+
```
Do we have test on quant_mode

huijuanzh commented on optimum/habana/diffusers/pipelines/flux/pipeline_flux_kontext.py:
```diff
@@ -0,0 +1,712 @@
+            )
+
+        # 1. Check inputs. Raise error if not correct
+        self.check_inputs(
```
https://github.com/huggingface/diffusers/blob/v0.35.2/src/diffusers/pipelines/flux/pipeline_flux_kontext.py#L888

huijuanzh commented on optimum/habana/diffusers/pipelines/flux/pipeline_flux_kontext.py:
```diff
@@ -0,0 +1,712 @@
+            pooled_prompt_embeds=pooled_prompt_embeds,
+            device=device,
+            num_images_per_prompt=num_images_per_prompt,
+            max_sequence_length=max_sequence_length,
```
https://github.com/huggingface/diffusers/blob/v0.35.2/src/diffusers/pipelines/flux/pipeline_flux_kontext.py#L937

huijuanzh replied:
Please also check if do_true_cfg work or is already split the native prompt into _split_inputs_into_batches.  https://github.com/huggingface/diffusers/blob/v0.35.2/src/diffusers/pipelines/flux/pipeline_flux_kontext.py#L939

huijuanzh commented on optimum/habana/diffusers/pipelines/flux/pipeline_flux_kontext.py:
```diff
@@ -0,0 +1,712 @@
+                        joint_attention_kwargs=self.joint_attention_kwargs,
+                        return_dict=False,
+                    )[0]
+                    noise_pred = noise_pred[:, : latents_batch.size(1)]
```
please check noise_pred with true_cfg_scale
https://github.com/huggingface/diffusers/blob/v0.35.2/src/diffusers/pipelines/flux/pipeline_flux_kontext.py#L1091

tianyuan211 committed:
Update pipeline_flux_kontext.py

auto resize to the closest aspect_ratio of the input_image; help with generation accuracy

tianyuan211 committed:
Update pipeline_flux_kontext.py

transformer should be changed to class attribute

tianyuan211 committed:
Update pipeline_flux_kontext.py

tianyuan211 committed:
Update pipeline_flux_kontext.py

tianyuan211 committed:
Update pipeline_flux_kontext.py

tianyuan211 committed:
Revert "Update pipeline_flux_kontext.py"

This reverts commit 066657fef3f33b183c124fac4ba539911dcf6945.

tianyuan211 committed:
Update pipeline_flux_kontext.py

tianyuan211 committed:
temp test

tianyuan211 committed:
Revert "temp test"

This reverts commit 4d2bbb329c4f271fd2a2093e7adfdd2293e59c90.

tianyuan211 committed:
Update pipeline_flux_kontext.py

tianyuan211 committed:
Merge branch 'aice/v1.22.0' into add-flux-kontext

tianyuan211 committed:
Update pipeline_flux_kontext.py

tianyuan211 committed:
Update pipeline_flux_kontext.py

tianyuan211 committed:
Update pipeline_flux_kontext.py

tianyuan211 committed:
Update pipeline_flux_kontext.py

tianyuan211 committed:
Update pipeline_flux_kontext.py

negative prompt functionally work

tianyuan211 committed:
Update pipeline_flux_kontext.py

tianyuan211 committed:
Update pipeline_flux_kontext.py

tianyuan211 committed:
Update pipeline_flux_kontext.py

tianyuan211 committed:
Merge branch 'aice/v1.22.0' into add-flux-kontext

tianyuan211 committed:
Update attention_processor.py

tianyuan211 committed:
Update attention_processor.py

tianyuan211 committed:
Merge branch 'aice/v1.22.0' into add-flux-kontext

tianyuan211 committed:
Update pipeline_flux_kontext.py

restore batch size feature

tianyuan211 committed:
Update pipeline_flux_kontext.py

tianyuan211 committed:
Create pipeline_flux2.py

temp test

tianyuan211 committed:
Update pipeline_flux2.py

tianyuan211 committed:
Delete pipeline_flux2.py

tianyuan211 committed:
Merge branch 'aice/v1.22.0' into add-flux-kontext

tianyuan211 committed:
Update image_to_image_generation.py

tianyuan211 committed:
Update image_to_image_generation.py
